### PR TITLE
DevHome SDK Changes For Developer Id

### PR DESCRIPTION
**Pull request by ssparach**
## Summary of the pull request
This PR makes additions to the DevHome SDK.

## Detailed description of the pull request / Additional comments
This PR adds enums that will be used to define the authentication user experience along with a method to retrieve the assigned values.  Additionally, this PR introduces new changes to event handling of Developer Id. 

A partner change will be created in the devhomegithubextension repo. 

## Validation steps performed
Local Build
Manual Testing of Sign-In/Sign-out flow

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: pluginsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl [hunk 1]
```diff
@@ -14,6 +14,20 @@ namespace Microsoft.Windows.DevHome.SDK
         DevDoctor = 5,
         Notifications = 6
     };
+    
+    [contract(Microsoft.Windows.DevHome.SDK.DevHomeContract, 1)]
```
Comment (dhoehna) [inline pluginsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl:18]: Please update the contract version to 2.

File: pluginsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl [hunk 2]
```diff
@@ -75,21 +89,29 @@ namespace Microsoft.Windows.DevHome.SDK
     [contract(Microsoft.Windows.DevHome.SDK.DevHomeContract, 1)]
     interface IDeveloperIdProvider
     {
+        AuthenticationExperienceKind GetAuthenticationExperienceKind(); 
+
         String GetName();
 
         IIterable<IDeveloperId> GetLoggedInDeveloperIds();
 
+        // This API is deprecated and should not be used. The API will be removed once all callers have been moved over.
```
Comment (dhoehna) [inline pluginsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl:98]: Is it a good idea to remove an API?  Would that be considered a breaking chage?

Comment (ssparach) [reply pluginsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl:98]: Resolved in offline thread

File: pluginsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl [hunk 3]
```diff
@@ -75,21 +89,29 @@ namespace Microsoft.Windows.DevHome.SDK
     [contract(Microsoft.Windows.DevHome.SDK.DevHomeContract, 1)]
     interface IDeveloperIdProvider
     {
```
Comment (manodasanW) [inline pluginsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl:91]: Do we want to add the new methods to the existing interface as technically that would be breaking?  Instead, typically you would create a new `IDeveloperIdProvider2` interface or something similar with the new contract version and add them there.  The new interface can require this interface to make it such that you only need to implement one.

File: pluginsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl [hunk 4]
```diff
@@ -75,21 +89,29 @@ namespace Microsoft.Windows.DevHome.SDK
...
+        // The below events (LoggedIn, LoggedOut and Updated) are legacy events.
         event Windows.Foundation.EventHandler<IDeveloperId> LoggedIn;
 
         event Windows.Foundation.EventHandler<IDeveloperId> LoggedOut;
 
         event Windows.Foundation.EventHandler<IDeveloperId> Updated;
+
+        event Windows.Foundation.TypedEventHandler<IDeveloperIdProvider, Object> Changed;
+
+        AuthenticationState developerIDState;
```
Comment (manodasanW) [inline pluginsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl:114]: Capitalize?  And I assume this is intentionally a get/set property?

File: pluginsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl (hunk 3 above)
Comment (ssparach) [reply pluginsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl:91]: The plan is to not update the contract version and take the breaking changes. Therefore, we can keep the new methods in the existing interface. Thanks!

**Review by dhoehna — APPROVED** [review]

File: pluginsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl (hunk 4 above)
Comment (ssparach) [reply pluginsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl:114]: Completing PR for now and resolving offline. I will be sure to take any changes in a future PR. Thanks!

**Review by manodasanW — APPROVED** [review]

**Pull request merged**